src/sentry/static/sentry/app/components/events/rrwebReplayer/rrWebReplayer.tsx

```diff
@@ -16,32 +33,68 @@ class RRWebReplayer extends React.Component<Props> {
 
   newRRWebPlayer: any;
 
-  async rrwebPlayer() {
+  rrwebPlayer = async () => {
+    this.setState({isLoading: true, hasError: false});
+
     const element = this.wrapperRef?.current;
 
     if (!element) {
       return;
     }
 
-    const {url} = this.props;
+    const {url, api} = this.props;
 
     try {
-      const resp = await fetch(url);
+      const resp = await api.requestPromise(url);
       const payload = await resp.json();
       this.newRRWebPlayer = new rrwebPlayer({
         target: element,
         autoplay: false,
         data: payload,
       });
+      this.setState({isLoading: false});
     } catch (err) {
       Sentry.captureException(err);
+      this.setState({isLoading: false, hasError: true});
     }
+  };
+
+  renderSubContent() {
+    const {hasError, isLoading} = this.state;
+
+    if (isLoading) {
+      return <LoadingIndicator />;
+    }
+
+    if (hasError) {
+      return (
+        <LoadingError
+          message={t('There was a problem loading the player')}
+          onRetry={this.rrwebPlayer}
+        />
+      );
+    }
+
+    return null;
```

Comment:
Can you show what this looks like? I think not having loading may have been intentional, but I'm not sure.

Reply:
yes sure... I believe this is a good idea because in case of an error we are displaying something like _____________ and it doesn't really help.

Reply:
![image](https://user-images.githubusercontent.com/29228205/98928794-7e938680-24da-11eb-8d4e-1c1e6defd647.png)

Reply:
I ended up removing the states as it was causing some issues... we might want to review this `rrWebPlayer` library afterwards

Reply:
I decided to remove the state as it was causing some problems. We might want to review this component afterward.